src/utils/logger.ts

```diff
@@ -1,20 +1,64 @@
-import Pino, { LoggerOptions } from "pino";
+import { createLogger, format, transports } from "winston";
 import { env } from "./env";
 
-const defaultOptions: LoggerOptions = {
-  redact: ["headers.authorization"],
-  transport: {
-    target: "pino-pretty",
-    options: {
-      translateTime: "HH:MM:ss Z",
-      ignore: "pid,hostname,reqId",
-      singleLine: true,
-    },
-  },
-  level: env.LOG_LEVEL,
+// Custom filter for stdout transport
+const filterOnlyInfoAndWarn = format((info) => {
+  if (info.level === "error") {
+    return false; // Exclude 'error' level logs
+  }
+  return info;
+});
+
+// Custom filter for stderr transport
+const filterOnlyErrors = format((info) => {
+  if (info.level !== "error") {
+    return false; // Exclude non-error level logs
+  }
+  return info;
+});
+
+const colorizeFormat = () => {
+  if (env.NODE_ENV === "development") {
+    return format.colorize();
+  } else {
+    return format.uncolorize();
+  }
 };
 
-const pino = Pino(defaultOptions);
+const winstonLogger = createLogger({
+  level: env.LOG_LEVEL,
+  format: format.combine(
+    format.timestamp(),
+    format.errors({ stack: true }),
```

Comment:
Does this show stack trace for all environments? Do we want it only for NODE_ENV=development?

Reply:
this will print the trace irrespective on the NODE_ENV but for API Response, we do not send the trace. The `errorHandler` middleware removes the trace when NODE_ENV is `production`

```
return reply.status(StatusCodes.BAD_REQUEST).send({
          error: {
            code: "BAD_REQUEST",
            message: errorObject.message ?? "Invalid Request",
            reason: errorObject ?? undefined,
            statusCode: 400,
            stack: env.NODE_ENV !== "production" ? _error.stack : undefined, //<---- Trace only send in response when NODE_ENV development
          },
});
```